detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/BitbakeExtractor.java

```diff
@@ -5,149 +5,174 @@
-            executableRunner,
-            bitbakeRecipesParser,
-            sourceDirectory,
-            buildEnvScript,
-            sourceArguments,
-            bash,
-            toolVersionLogger,
-            buildFileFinder,
-            bitbakeEnvironmentParser
-        );
-        bitbakeSession.logBitbakeVersion();
-        File buildDir = bitbakeSession.determineBuildDir();
-        BitbakeEnvironment bitbakeEnvironment = bitbakeSession.executeBitbakeForEnvironment();
-        ShowRecipesResults showRecipesResults = bitbakeSession.executeBitbakeForRecipeLayerCatalog();
-        for (String targetImage : packageNames) {
-            try {
-                codeLocations.add(generateCodeLocationForTargetImage(
-                    followSymLinks,
-                    searchDepth,
-                    dependencyTypeFilter,
-                    bitbakeSession,
-                    buildDir,
-                    bitbakeEnvironment,
-                    showRecipesResults,
-                    targetImage
-                ));
-            } catch (IOException | IntegrationException | NotImplementedException | ExecutableFailedException e) {
-                logger.error(String.format("Failed to extract a Code Location while running Bitbake against package '%s': %s", targetImage, e.getMessage()));
-                logger.debug(e.getMessage(), e);
-            }
-        }
-        Extraction extraction;
+    public Extraction extract(File sourceDirectory, ExecutableTarget bashExecutable, File buildEnvironmentFile) throws ExecutableFailedException, IOException {
+        toolVersionLogger.log(() -> bitbakeCommandRunner.runBitbakeVersion(sourceDirectory, bashExecutable, buildEnvironmentFile));
+
+        File buildDirectory = determineBuildDirectory(sourceDirectory, bashExecutable, buildEnvironmentFile);
+        BitbakeEnvironment bitbakeEnvironment = executeBitbakeForEnvironment(sourceDirectory, bashExecutable, buildEnvironmentFile);
+        ShowRecipesResults showRecipesResults = executeBitbakeForRecipeLayerCatalog(sourceDirectory, bashExecutable, buildEnvironmentFile);
+
+        List<CodeLocation> codeLocations = packageNames.stream()
+            .map(targetPackage -> {
+                try {
+                    return generateCodeLocationForTargetPackage(
+                        sourceDirectory,
+                        bashExecutable,
+                        buildEnvironmentFile,
+                        buildDirectory,
+                        targetPackage,
+                        showRecipesResults,
+                        bitbakeEnvironment
+                    );
+                } catch (ExecutableFailedException | IntegrationException | IOException e) {
+                    logger.error("Failed to extract a Code Location while running Bitbake against package '{}': {}", targetPackage, e.getMessage());
+                    logger.debug(e.getMessage(), e);
+                    return null;
+                }
+            })
```

Comment:
Good idea. Instead of allowing allowing `generateCodeLocationForTargetPackage` to throw an exception, I moved the try-catch into there and made the method return an `Optional<CodeLocation>`
I think it significantly reduces the complexity of `BitbakeExtractor::extract`